services/incentive/src/main/java/org/cryptimeleon/incentive/services/incentive/repository/OfflineDSPRepository.java

```diff
@@ -33,4 +34,12 @@ public interface OfflineDSPRepository {
      * Returns true if and only if the double-spending database contains a node for the passed dsid.
      */
     boolean containsDsid(GroupElement dsid);
+
+    void addShortWaitPeriod();
+
+    void addLongWaitPeriod();
+
+    void removeAllWaitPeriod();
+
+    LocalDateTime getWaitUntil();
```

Comment:
why merging the offlinedsprepo and cyclingscheduler classes?

Reply:
Makes it easier to mock an instance implementing the interface. And we never used the tow interfaces individually, so I decided to simplify the code a little bit :)